e2e-app/src/app/typeahead/focus/typeahead-focus.pw-spec.ts

```diff
@@ -1,57 +1,64 @@
-import {waitForFocus, Key, openUrl, sendKey} from '../../tools.pw-po';
+import {waitForFocus, Key, openUrl, sendKey, timeoutMessage} from '../../tools.pw-po';
 import {test} from '../../../../playwright.conf';
-import {getTypeaheadValue, SELECTOR_TYPEAHEAD, SELECTOR_TYPEAHEAD_ITEMS, SELECTOR_TYPEAHEAD_WINDOW} from '../typeahead';
+import {SELECTOR_TYPEAHEAD, SELECTOR_TYPEAHEAD_ITEMS, SELECTOR_TYPEAHEAD_WINDOW} from '../typeahead';
 
 describe('Typeahead', () => {
 
-  const expectTypeaheadFocused = async() => await waitForFocus(SELECTOR_TYPEAHEAD, `Typeahead should be focused`);
+  const waitForTypeaheadFocused = async() => await waitForFocus(SELECTOR_TYPEAHEAD, `Typeahead should be focused`);
 
-  const expectDropdownOpen = async(suggestions = 10) => {
-    const items = await test.page.$$(SELECTOR_TYPEAHEAD_ITEMS);
-    expect(items.length).toBe(suggestions, `Wrong number of suggestions`);
+  const waitForDropdownOpen = async(suggestions = 10) => {
+    await timeoutMessage(
+        test.page.waitForFunction(
+            (args) => (document.querySelectorAll(args[0]).length === args[1]),
+            [SELECTOR_TYPEAHEAD_ITEMS, suggestions] as[string, number]),
+        `Wrong number of suggestions (expected ${suggestions})`);
```

Comment:
I think you can 'lighten' the code by using a string function, i.e.:
```
test.page.waitForFunction(`document.querySelectorAll(${SELECTOR_TYPEAHEAD_ITEMS}).length === ${number}`)
```

What do you think ?

Reply:
Indeed